### ISSUE TITLE
Google Drive: tls: bad record MAC

### ISSUE THREAD

**Issue by PedroC1999** [rclone/rclone#3338]
Firstly I would like to say that I have viewed the similar issue reports and have tried the debugging methods suggested (i.e the disabling of checksums)

I am getting this issue when copying large files from my Raspberry Pi 4 (4gb) running the latest and fully updated Raspbian Buster Lite.

I am attempting to copy files to a Team Drive. I get this issue whether I use rclone copy or do a regular copy command to the rclone mount.

The issue comes up only after 1 or 2 large transfers, I can normally upload 1 or 2 files but then each large file after that will fail. After a prolonged period of time, if I try the upload again, it will succeed.

The files are between 1 and 2gb, but anything larger suffers from the same issue.

The file fails to upload with the "tls: bad record MAC" error, retries 3 times (which fail with the same error each time) then cancels the upload.

I am able to provide any logs and try any debugging steps provided.

**Comment by animosity22** [comment]
You should follow the bug template and provide version / logs / operating system / all the fields that are there when you report an issue.

**Comment by PedroC1999** [comment]
I do apologise, I wasn't sure if it directly qualified as a "bug" as it felt like more of an embedded, issue so I used the regular issue option.

I did miss the rclone version info request, which is as follows. I stated operating system within the description (Raspbian Buster)

rclone v1.48.0
- os/arch: linux/arm
- go version: go1.12.6

I am running a copy command now with the -vv flag and will attach it soon

**Comment by PedroC1999** [comment]
The -vv output of copying a ~1.6gb file created using `dd if=/dev/zero of=file.txt count=1024 bs=1648576` for testing purposes

`pi@RPI4B:~ $ sudo rclone copy file.txt "entertainment:/Temp Upload" --progress -vv
2019/07/10 14:53:30 DEBUG : rclone: Version "v1.48.0" starting with parameters ["rclone" "copy" "file.txt" "entertainment:/Temp Upload" "--progress" "-vv"]
2019/07/10 14:53:30 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2019-07-10 14:53:32 DEBUG : file.txt: Couldn't find file - need to transfer
2019-07-10 14:53:33 DEBUG : file.txt: Sending chunk 0 length 8388608
2019-07-10 14:53:38 DEBUG : file.txt: Sending chunk 8388608 length 8388608
2019-07-10 14:53:42 DEBUG : file.txt: Sending chunk 16777216 length 8388608
2019-07-10 14:53:47 DEBUG : file.txt: Sending chunk 25165824 length 8388608
2019-07-10 14:53:51 DEBUG : file.txt: Sending chunk 33554432 length 8388608
2019-07-10 14:53:55 DEBUG : file.txt: Sending chunk 41943040 length 8388608
2019-07-10 14:54:00 DEBUG : file.txt: Sending chunk 50331648 length 8388608
2019-07-10 14:54:04 DEBUG : file.txt: Sending chunk 58720256 length 8388608
2019-07-10 14:54:09 DEBUG : file.txt: Sending chunk 67108864 length 8388608
2019-07-10 14:54:13 DEBUG : file.txt: Sending chunk 75497472 length 8388608
2019-07-10 14:54:17 DEBUG : file.txt: Sending chunk 83886080 length 8388608
2019-07-10 14:54:23 DEBUG : file.txt: Sending chunk 92274688 length 8388608
2019-07-10 14:54:25 ERROR : file.txt: Failed to copy: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2UqJHowbEuZsqHtgqfFg0s8UDW_lSL7bFoTAA0KTiEINo7C2brpJYg4Qhak9BLvyE4nrSuK4PPMR9h6DOZnYSnBgZikigA: remote error: tls: bad record MAC
2019-07-10 14:54:25 ERROR : Attempt 1/3 failed with 2 errors and: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2UqJHowbEuZsqHtgqfFg0s8UDW_lSL7bFoTAA0KTiEINo7C2brpJYg4Qhak9BLvyE4nrSuK4PPMR9h6DOZnYSnBgZikigA: remote error: tls: bad record MAC
2019-07-10 14:54:26 DEBUG : file.txt: Couldn't find file - need to transfer
2019-07-10 14:54:27 DEBUG : file.txt: Sending chunk 0 length 8388608
2019-07-10 14:54:35 DEBUG : file.txt: Sending chunk 8388608 length 8388608
2019-07-10 14:54:40 DEBUG : file.txt: Sending chunk 16777216 length 8388608
2019-07-10 14:54:47 DEBUG : file.txt: Sending chunk 25165824 length 8388608
2019-07-10 14:54:56 DEBUG : file.txt: Sending chunk 33554432 length 8388608
2019-07-10 14:55:01 ERROR : file.txt: Failed to copy: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2Ur-Bz4BJxKZUR83y6ldpO1-TuOuhjX0JX5H_LG8U9Eh3Di39CL1hv7ORuK0ZNZUmBckIkjaLFGlmG038YeuFH2KtQsAJoh_oAoV4rYzN8QfI5FJMs0: remote error: tls: bad record MAC
2019-07-10 14:55:01 ERROR : Attempt 2/3 failed with 2 errors and: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2Ur-Bz4BJxKZUR83y6ldpO1-TuOuhjX0JX5H_LG8U9Eh3Di39CL1hv7ORuK0ZNZUmBckIkjaLFGlmG038YeuFH2KtQsAJoh_oAoV4rYzN8QfI5FJMs0: remote error: tls: bad record MAC
2019-07-10 14:55:01 DEBUG : file.txt: Couldn't find file - need to transfer
2019-07-10 14:55:02 DEBUG : file.txt: Sending chunk 0 length 8388608
2019-07-10 14:55:09 DEBUG : file.txt: Sending chunk 8388608 length 8388608
2019-07-10 14:55:13 DEBUG : file.txt: Sending chunk 16777216 length 8388608
2019-07-10 14:55:19 DEBUG : file.txt: Sending chunk 25165824 length 8388608
2019-07-10 14:55:27 DEBUG : file.txt: Sending chunk 33554432 length 8388608
2019-07-10 14:55:27 ERROR : file.txt: Failed to copy: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2Uo5UXlR-IBDrspgFoXS3fQLcfewRm8zHlg16f7uFp0V2r0frPyyrs_YKkVuYIU7ShUCWHX3EG-cdC8sKEZ-GMSOi7M8zA: remote error: tls: bad record MAC
2019-07-10 14:55:27 ERROR : Attempt 3/3 failed with 2 errors and: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2Uo5UXlR-IBDrspgFoXS3fQLcfewRm8zHlg16f7uFp0V2r0frPyyrs_YKkVuYIU7ShUCWHX3EG-cdC8sKEZ-GMSOi7M8zA: remote error: tls: bad record MAC
Transferred:      161.242M / 161.242 MBytes, 100%, 1.382 MBytes/s, ETA 0s
Errors:                 2 (retrying may help)
Checks:                 0 / 0, -
Transferred:            0 / 0, -
Elapsed time:     1m56.6s
2019/07/10 14:55:27 Failed to copy with 2 errors: last error was: Post https://www.googleapis.com/upload/drive/v3/files?alt=json&fields=id%2Cname%2Csize%2Cmd5Checksum%2Ctrashed%2CmodifiedTime%2CcreatedTime%2CmimeType%2Cparents%2CwebViewLink&supportsTeamDrives=true&uploadType=resumable&upload_id=AEnB2Uo5UXlR-IBDrspgFoXS3fQLcfewRm8zHlg16f7uFp0V2r0frPyyrs_YKkVuYIU7ShUCWHX3EG-cdC8sKEZ-GMSOi7M8zA: remote error: tls: bad record MAC`

**Comment by animosity22** [comment]
There are a few duplicates of the item:

https://github.com/ncw/rclone/issues/1893

https://github.com/ncw/rclone/issues/1893

Most point to an OS related thing but I'm not very familiar with PI or how to upgrade/validate those things.

**Comment by PedroC1999** [comment]
I did state that I had tried troubleshooting steps described in other similar issues, but was not successful in fixing the issue

**Comment by ncw** [comment]
How repeatable is the problem @PedroC1999 ?

What that error means is that a TLS packet got corrupted somewhere.  That can be on the network, in memory, in the driver or it could be a bug in the rclone runtime.

In my experience it is mostly likely to be the network driver - can you try different versions of that?

**Comment by PedroC1999** [comment]
It's very highly repeatable, inevitable is a better word for it.

I am unsure on how to do that on the RPI, but I will see what I can do

**Comment by ncw** [comment]
> It's very highly repeatable, inevitable is a better word for it.

Attached is the latest version of rclone built with the latest version of go.  Can you see if that makes a difference?

[rclone-arm-linux-v1.48.0-073-g266600db-beta.gz](https://github.com/ncw/rclone/files/3421877/rclone-arm-linux-v1.48.0-073-g266600db-beta.gz)

If it is a go bug then repeatable is good!

> I am unsure on how to do that on the RPI, but I will see what I can do

Can you try WiFi vs wired ethernet or something like that?

**Comment by mbeckford88** [comment]
Hi all. Have tried the latest beta version, still getting quite a few of these errors on large files. I'm getting this on W10 using the amd64 version however.

**Comment by ncw** [comment]
As a workaround I've made the error retriable here.  This means that should this error occur transiently rclone will retry the chunk it is uploading and carry on.

Can everyone affected give it a test and report back?

https://beta.rclone.org/branch/v1.48.0-103-gf909f191-fix-3338-bad-record-mac-beta/ (uploaded in 15-30 mins)

**Comment by mbeckford88** [comment]
Looks good from my side (W10/amd64). Not seeing the TLS error flag against large files now.

**Comment by ncw** [comment]
Great :-)

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.49
